### ISSUE TITLE
submit task to OpenPAI cluster failed

### ISSUE THREAD

**Issue by guoxiaojie-schinper** [microsoft/nni#4022]
**Describe the issue**:

I have deployed a PAI platform, and everything is ok if I submit the task through PAI-Web. But when I used NNI to create a task, it always fails. My config content is as follow:

```
searchSpaceFile: search_space.json
trialCommand: python3 mnist.py
trialGpuNumber: 0
trialConcurrency: 4
maxTrialNumber: 10
nniManagerIp: 172.18.18.206
tuner:
  name: TPE
  classArgs:
    optimize_mode: maximize
trainingService:
  platform: openpai
  host: https://172.18.18.206
  username: admin
  token: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6ImFkbWluIiwiYXBwbGljYXRpb24iOmZhbHNlLCJqb2JTcGVjaWZpYyI6ZmFsc2UsImVuY29kZWRGcmFtZXdvcmtOYW1lIjoiIiwiaWF0IjoxNjI4MDcxNjQxLCJleHAiOjE2Mjg2NzY0NDF9.8UkAX_Rf08V3bPeijoJKf5fguVt6RCeB6Un_qoDwAlE
  dockerImage: msranni/nni
  trialCpuNumber: 4
  trialMemorySize: 8GB
  storageConfigName: nfs-data
  localStorageMountPoint: /schinper-nni
  containerStorageMountPoint: /mnt/nfs-storage
```

**Environment**:
- NNI version: 2.3
- Training service (local|remote|pai|aml|etc): pai
- Client OS: Ubuntu20.04
- Server OS (for remote mode only): Ubuntu20.04
- Python version: Python 3.7.6
- PyTorch/TensorFlow version:
- Is conda/virtualenv/venv used?: virtualenv
- Is running in Docker?:

**Log message**:

 - nnimanager.log:

```
[2021-08-04 21:03:59] ERROR (NNIManager) TypeError: this.internalJoin is not a function
    at StorageService.joinPath (/opt/virtualenvs/nni/lib/python3.7/site-packages/nni_node/training_service/reusable/storageService.js:127:29)
    at TrialDispatcher.run (/opt/virtualenvs/nni/lib/python3.7/site-packages/nni_node/training_service/reusable/trialDispatcher.js:187:43)
[2021-08-04 21:03:59] ERROR (NNIExperimentsManager) Error: File has been locked.
    at Object.withLockSync (/opt/virtualenvs/nni/lib/python3.7/site-packages/nni_node/common/utils.js:364:19)
    at NNIExperimentsManager.withLockSync (/opt/virtualenvs/nni/lib/python3.7/site-packages/nni_node/core/nniExperimentsManager.js:91:24)
    at NNIExperimentsManager.setExperimentInfo (/opt/virtualenvs/nni/lib/python3.7/site-packages/nni_node/core/nniExperimentsManager.js:55:18)
    at NNIManager.setEndtime (/opt/virtualenvs/nni/lib/python3.7/site-packages/nni_node/core/nnimanager.js:705:32)
    at NNIManager.logError (/opt/virtualenvs/nni/lib/python3.7/site-packages/nni_node/core/nnimanager.js:693:14)
    at NNIManager.criticalError (/opt/virtualenvs/nni/lib/python3.7/site-packages/nni_node/core/nnimanager.js:685:14)
    at run.catch (/opt/virtualenvs/nni/lib/python3.7/site-packages/nni_node/core/nnimanager.js:138:18)
[2021-08-04 21:03:59] INFO (NNIManager) Change NNIManager status from: RUNNING to: ERROR
[2021-08-04 21:03:59] ERROR (NNIExperimentsManager) Error: File has been locked.
    at Object.withLockSync (/opt/virtualenvs/nni/lib/python3.7/site-packages/nni_node/common/utils.js:364:19)
    at NNIExperimentsManager.withLockSync (/opt/virtualenvs/nni/lib/python3.7/site-packages/nni_node/core/nniExperimentsManager.js:91:24)
    at NNIExperimentsManager.setExperimentInfo (/opt/virtualenvs/nni/lib/python3.7/site-packages/nni_node/core/nniExperimentsManager.js:55:18)
    at NNIManager.setStatus (/opt/virtualenvs/nni/lib/python3.7/site-packages/nni_node/core/nnimanager.js:700:36)
    at NNIManager.logError (/opt/virtualenvs/nni/lib/python3.7/site-packages/nni_node/core/nnimanager.js:694:14)
    at NNIManager.criticalError (/opt/virtualenvs/nni/lib/python3.7/site-packages/nni_node/core/nnimanager.js:685:14)
    at run.catch (/opt/virtualenvs/nni/lib/python3.7/site-packages/nni_node/core/nnimanager.js:138:18)

```

 - dispatcher.log:
```
[2021-08-04 21:03:59] INFO (hyperopt.tpe/Thread-1) tpe_transform took 0.003151 seconds
[2021-08-04 21:03:59] INFO (hyperopt.tpe/Thread-1) TPE using 0 trials
[2021-08-04 21:03:59] INFO (hyperopt.tpe/Thread-1) tpe_transform took 0.002367 seconds
[2021-08-04 21:03:59] INFO (hyperopt.tpe/Thread-1) TPE using 0 trials
[2021-08-04 21:03:59] INFO (hyperopt.tpe/Thread-1) tpe_transform took 0.002332 seconds
[2021-08-04 21:03:59] INFO (hyperopt.tpe/Thread-1) TPE using 0 trials
[2021-08-04 21:03:59] INFO (hyperopt.tpe/Thread-1) tpe_transform took 0.002332 seconds
[2021-08-04 21:03:59] INFO (hyperopt.tpe/Thread-1) TPE using 0 trials
```
 - nnictl stdout and stderr:
```
-----------------------------------------------------------------------
                Experiment start time 2021-08-04 21:03:58
-----------------------------------------------------------------------
{ TypeError: this.internalJoin is not a function
    at StorageService.joinPath (/opt/virtualenvs/nni/lib/python3.7/site-packages/nni_node/training_service/reusable/storageService.js:127:29)
    at TrialDispatcher.run (/opt/virtualenvs/nni/lib/python3.7/site-packages/nni_node/training_service/reusable/trialDispatcher.js:187:43)
  name: '',
  cause:
   TypeError: this.internalJoin is not a function
       at StorageService.joinPath (/opt/virtualenvs/nni/lib/python3.7/site-packages/nni_node/training_service/reusable/storageService.js:127:29)
       at TrialDispatcher.run (/opt/virtualenvs/nni/lib/python3.7/site-packages/nni_node/training_service/reusable/trialDispatcher.js:187:43) }
```
 

**How to reproduce it?**:

Running command as the following,

```
nnictl create -c demo.yaml --port 1500
```

**Comment by guoxiaojie-schinper** [comment]
I have checked the latest ts code in NNI project, [nnimanager.ts](https://github.com/microsoft/nni/blob/master/ts/nni_manager/core/nnimanager.ts). In the initTrainingService() function, I didn't see NNI import paiTrainingService.  Why?

```
public async setClusterMetadata(key: string, value: string): Promise<void> {
        // Hack for supporting v2 config, need refactor
        if (this.trainingService === undefined) {
            this.log.info('Setup training service...');
            switch (key) {
                case 'kubeflow_config': {
                    const kubeflowModule = await import('../training_service/kubernetes/kubeflow/kubeflowTrainingService');
                    this.trainingService = new kubeflowModule.KubeflowTrainingService();
                    break;
                }
                case 'frameworkcontroller_config': {
                    const fcModule = await import('../training_service/kubernetes/frameworkcontroller/frameworkcontrollerTrainingService');
                    this.trainingService = new fcModule.FrameworkControllerTrainingService();
                    break;
                }
                case 'adl_config': {
                    const adlModule = await import('../training_service/kubernetes/adl/adlTrainingService');
                    this.trainingService = new adlModule.AdlTrainingService();
                    break;
                }
                default:
                    throw new Error("Setup training service failed.");
            }
        }
        await this.trainingService.setClusterMetadata(key, value);
    }
```

**Comment by SparkSnail** [comment]
PAI has enabled reuse mode, this is a bug in latest code. I've created a pr to fix this bug https://github.com/microsoft/nni/pull/4027.   
You can follow the pr to change your local code, or wait for the pr to merge into master branch and sync with master branch code, and install NNI from source code to mitigate this issue. This bug fix will merge into next NNI release in recent days.

**Comment by guoxiaojie-schinper** [comment]
> PAI has enabled reuse mode, this is a bug in latest code. I've created a pr to fix this bug #4027.
> You can follow the pr to change your local code, or wait for the pr to merge into master branch and sync with master branch code, and install NNI from source code to mitigate this issue. This bug fix will merge into next NNI release in recent days.

Thanks for your reply. I look forward to the new release.

**Comment by DC-y** [comment]
Hi,  @guoxiaojie-schinper ,
I have also used the latest coe,  met the same error, after followed you step and  changed the local code,  reintall NNI, but the error still exist.  I think that the bug have not been fix completely .

**Comment by Louis-J** [comment]
fixed in #4027.
